src/MongoDB/Manager.c

```diff
@@ -279,7 +279,9 @@ PHP_METHOD(Manager, selectServer)
 	}
 
 	readPreference = phongo_read_preference_from_zval(zreadPreference TSRMLS_CC);
-	selected_server = mongoc_topology_select(intern->client->topology, MONGOC_SS_READ, readPreference, MONGOC_SS_DEFAULT_LOCAL_THRESHOLD_MS, &error);
+	/* TODO: Remove (mongoc_read_prefs_t *) casting once
+	 * https://github.com/mongodb/mongo-c-driver/pull/315 is merged. */
+	selected_server = mongoc_client_select_server(intern->client, true, (mongoc_read_prefs_t *) readPreference, &error);
```

Comment:
When I implement a similar thing for HHVM, I get tests failing. For example "tests/functional/cursor-001.phpt" fails with:

```
Fatal error: Uncaught exception 'MongoDB\Driver\Exception\ConnectionTimeoutException' with message 'Cannot use read preferences with for_writes = true' in /home/derick/dev/php/derickr-mongo-php-driver/tests/utils/tools.php:106
Stack trace:
#0 /home/derick/dev/php/derickr-mongo-php-driver/tests/utils/tools.php(106): MongoDB\Driver\Manager->selectServer()
#1 /home/derick/dev/php/derickr-mongo-php-driver/tests/functional/cursor-001.phpt(5): LOAD()
```

The "true" is the second argument to client_select_server, so I don't think this is the right fix. I am curious as to why this does not fail for you though.